extensions/pkg/controller/worker/genericactuator/actuator_delete.go

```diff
@@ -68,25 +69,19 @@ func (a *genericActuator) Delete(ctx context.Context, worker *extensionsv1alpha1
 		return errors.Wrapf(err, "marking all machines for forceful deletion failed")
 	}
 
-	// Get the list of all existing machine deployments.
-	existingMachineDeployments := &machinev1alpha1.MachineDeploymentList{}
-	if err := a.client.List(ctx, existingMachineDeployments, client.InNamespace(worker.Namespace)); err != nil {
-		return err
-	}
-
 	// Delete all machine deployments.
-	if err := a.cleanupMachineDeployments(ctx, logger, existingMachineDeployments, nil); err != nil {
-		return errors.Wrapf(err, "cleaning up machine deployments failed")
+	if err := a.client.DeleteAllOf(ctx, &machinev1alpha1.MachineDeployment{}, client.InNamespace(worker.Namespace)); err != nil {
```

Comment:
Can you add back some logging for these steps?
```suggestion
	logger.Info("Deleting all MachineDeployments")
	if err := a.client.DeleteAllOf(ctx, &machinev1alpha1.MachineDeployment{}, client.InNamespace(worker.Namespace)); err != nil {
```